frontend/src/features/admin-form/share/ShareFormModal.tsx

```diff
@@ -201,7 +208,17 @@ export const ShareFormModal = ({
       return
     } catch (err) {
       setClaimGoLoading(false)
-      setGoLinkHelperText(goLinkClaimFailureHelperText)
+
+      let errMessage =
+        'Something went wrong. Try refreshing this page. If this issue persists, contact support@form.gov.sg.'
+
+      if (err instanceof HttpError && err.code === StatusCodes.BAD_REQUEST)
+        errMessage =
+          err.message === GO_VALIDATION_ERROR_MESSAGE
+            ? 'Short links should only consist of lowercase letters, numbers and hyphens.'
+            : 'Short link is already in use.'
```

Comment:
Could you refactor this as a case switch for it to be cleaner? So err.message should be `GO_VALIDATION_ERROR_MESSAGE | GO_ALREADY_EXIST_ERROR_MESSAGE` typed

Reply:
[b9ebcd9](https://github.com/opengovsg/FormSG/pull/6714/commits/b9ebcd948a2395f29eb5cac71b4f32f4037c7535)